odc/stac/eo3/_eo3converter.py

```diff
@@ -313,7 +321,12 @@ def stac2ds(
 
     """
     products: Dict[str, DatasetType] = {} if product_cache is None else product_cache
-    for item in items:
+    if asset_urls is None:
+        asset_urls = []
+
+    items_with_urls = itertools.zip_longest(items, asset_urls)
+
+    for item, url in items_with_urls:
```

Comment:
Anything being passed `item` will need to account for the possibility of it being `None`.